core/server/common/src/main/java/alluxio/master/journal/DefaultJournalMaster.java

```diff
@@ -26,21 +29,29 @@
 public class DefaultJournalMaster implements JournalMaster {
   private final JournalDomain mJournalDomain;
   private final JournalSystem mJournalSystem;
+  private final PrimarySelector mPrimarySelector;
 
   /**
    * @param journalDomain domain for the journal
    * @param journalSystem internal {@link JournalSystem} reference
+   * @param primarySelector the primary selector to get the node state
    */
-  public DefaultJournalMaster(JournalDomain journalDomain, JournalSystem journalSystem) {
+  public DefaultJournalMaster(
+      JournalDomain journalDomain, JournalSystem journalSystem, PrimarySelector primarySelector) {
     mJournalDomain = journalDomain;
     mJournalSystem = journalSystem;
+    mPrimarySelector = primarySelector;
   }
 
   private void checkQuorumOpSupported() {
     if (!(mJournalSystem instanceof RaftJournalSystem)) {
       throw new UnsupportedOperationException(
           "Quorum operations are supported for journal type: EMBEDDED");
     }
+    if (!((RaftJournalSystem) (mJournalSystem)).isLeader()) {
+      throw new UnsupportedOperationException(
+          "Quorum operations are only supported on Raft leader");
+    }
```

Comment:
I feel quorum operations should not happen on standbys. This isn't an issue today because standby service doesn't run grpc server but adding this check doesn't do harm. wdyt? I'm open to removing this.

Reply:
This lgtm